src/AppInitializer.tsx

```diff
@@ -33,14 +43,14 @@ class AppInitializer extends React.Component<Props, AppInitializerState> {
     )
   }
 
-  componentDidMount() {
-    if (document.title !== docTitleConfig.name) {
-      document.title = docTitleConfig.name
-    }
-    document
-      .querySelector('meta[name="description"]')!
-      .setAttribute('content', docTitleConfig.description)
+  resetSession = () => {
+    SynapseClient.signOut(this.getSession)
+    this.setState({
+      showLoginDialog: false,
+    })
+  }
 
+  getSession = () => {
```

Comment:
Should this return a promise?

Reply:
I just made a commit to flatten out the function with async/await and also consolidated the logic to be easier to read, thanks for pointing that out! 

I couldn't figure out how to make this function return as a promise though. Since this function is called by synapseclient, I'd like to avoid synapseclient working any harder by calling this function as a promise, but I did add a helper function initAnonymousUserState which I think better pieces out this function.